nsuns-server/src/auth/token.rs

```diff
@@ -0,0 +1,145 @@
+use anyhow::anyhow;
+use chrono::{serde::ts_milliseconds, DateTime, Days, Utc};
+use http::StatusCode;
+use jsonwebtoken::{
+    decode, encode,
+    errors::{Error, ErrorKind},
+    DecodingKey, EncodingKey, Header, Validation,
+};
+use secrecy::ExposeSecret;
+use serde::{Deserialize, Serialize};
+use sqlx::{Database, Decode, Encode, Type};
+use time::Duration;
+use tower_cookies::{cookie::SameSite, Cookie};
+use uuid::Uuid;
+
+use crate::{db::DB, error::ErrorWithStatus};
+
+use super::settings::AuthSettings;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Claims {
+    pub owner_id: OwnerId,
+    pub user_id: Option<Uuid>,
+    #[serde(with = "ts_milliseconds")]
+    pub exp: DateTime<Utc>,
+}
+
+impl Claims {
+    pub fn generate(owner_id: Uuid, user_id: Option<Uuid>) -> Self {
+        Self {
+            owner_id: OwnerId(owner_id),
+            user_id,
+            exp: create_new_expiry_date(),
+        }
+    }
+}
+
+/// The authenticated resource owner id
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub struct OwnerId(Uuid);
```

Comment:
Using a newtype pattern here to avoid accidentally checking the wrong id. It also restricts creation of this type to come from a request only.